### DISCUSSION TITLE
Strongswan starts as a initiator problem

### DISCUSSION THREAD

**Discussion by joseph1267** [strongswan/strongswan#666] [Q&A] [answered]
Hi Strongswan Team, 
I have encountered a issue related to the initiator of the IKE message. 
In my setup, Strongswan is the initiator of the IKE message and the SA_INIT request is being sent and response is being received. Later on, after the IPsec tunnel establishment, the free5gc (N3IWF) initiates Create_Child_SA message to the strongswan. But, strongswan is having issue with processing the Create_Child_SA request message. I have been going through the strongswan code, and found out that there is a initiator and responder flag issue. In ike_sa_manager.c file, during the checkout_by_message function, during the get_entry_by_id, it checks the initiator and responder flags, and because Strongswan was the initiator for the SA_INIT message, when free5gc initiates the Create_Child_SA, it gets an error. So, my question is that, does the strongswan keeps on being a initiator state until the end of the whole exchange? or is there a "creared" state that strongswan goes back to?
![image](https://user-images.githubusercontent.com/45336417/137213755-dd5afdf0-a5a2-48c3-b786-1df3fe45ac18.png)

Thank you in advance

**Comment by tobiasbrunner** [comment, marked as answer]
> So, my question is that, does the strongswan keeps on being a initiator state until the end of the whole exchange?

If you mean the IKE_SA with "whole exchange" then yes. The _Initiator_ flag is set in messages sent by the original initiator of the IKE_SA, not the initiator of a particular exchange (see [RFC 7296, section 3.1](https://datatracker.ietf.org/doc/html/rfc7296#section-3.1)). The only way this can change from one peer to the other is during an IKE_SA rekeying initiated by the other peer. But that replaces the IKE_SA, so for each IKE_SA the initiator (or original initiator) state always stays the same.

If the other implementation does this incorrectly, you have to fix it there.
